core/src/main/scala/geotrellis/store/query/QueryF.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.store.query
+
+import geotrellis.vector.ProjectedExtent
+
+import io.circe._
+import io.circe.syntax._
+import io.circe.generic.JsonCodec
+import cats.Functor
+import cats.syntax.either._ // scala 2.11 compact
+import higherkindness.droste.scheme
+import higherkindness.droste.{Algebra, Coalgebra}
+import higherkindness.droste.syntax.fix._
+
+import java.time.ZonedDateTime
+
+@JsonCodec sealed trait QueryF[A]
+
+object QueryF {
+  // contains java8.time codecs for scala 2.11
+  import CirceCompat._
+
+  /** Tree leaves */
+  @JsonCodec case class Or[A](left: A, right: A) extends QueryF[A]
+  @JsonCodec case class And[A](left: A, right: A) extends QueryF[A]
+  @JsonCodec case class Intersects[A](projectedExtent: ProjectedExtent) extends QueryF[A]
+  @JsonCodec case class Contains[A](projectedExtent: ProjectedExtent) extends QueryF[A]
+  @JsonCodec case class Covers[A](projectedExtent: ProjectedExtent) extends QueryF[A]
+  @JsonCodec case class At[A](time: ZonedDateTime, fieldName: String = "time") extends QueryF[A]
+  @JsonCodec case class Between[A](from: ZonedDateTime, to: ZonedDateTime, fieldName: String = "time") extends QueryF[A]
+  @JsonCodec case class WithName[A](name: String) extends QueryF[A]
+  @JsonCodec case class WithNames[A](names: Set[String]) extends QueryF[A]
+  @JsonCodec case class Nothing[A]() extends QueryF[A]
```

Comment:
Perhaps `None` is a better complement to `All`? `Nothing` seems to me like literally 'no thing' whereas this is just none of the instances of w.e particular thing we're talking about

Reply:
@moradology `None` is a reserved word; and the `none` function potentially can confuse users. I'm happy to try Any / Nothing though but probably `any` can also confuse people as a part of a Query DSL

Reply:
:yoda: